azurerm/internal/services/postgres/postgresql_aad_administrator_resource.go

```diff
@@ -73,6 +74,15 @@ func resourcePostgreSQLAdministratorCreateUpdate(d *schema.ResourceData, meta in
 	serverName := d.Get("server_name").(string)
 	resGroup := d.Get("resource_group_name").(string)
 	login := d.Get("login").(string)
+	disallowedLogins := [7]string{"azure_superuser", "azure_pg_admin", "admin", "administrator", "root", "guest", "public"}
+	for _, v := range disallowedLogins {
+		if v == login {
+			return fmt.Errorf("Error - PostgreSQL AD Administrator login can not be %q. (Resource Group %q, Server %q)", login, resGroup, serverName)
+		}
+	}
+	if strings.HasPrefix(login, "pg_") {
+		return fmt.Errorf("Error - PostgreSQL AD Administrator login can not start with 'pg_'. (Resource Group %q, Server %q)", resGroup, serverName)
+	}
```

Comment:
instead of doing this in the create/update function could we add a validation function in the schema?

Reply:
Hey - have tried adding a validation function, though could have got it completely wrong so feel free to correct me 😄